app/services/work_history_with_breaks.rb

```diff
@@ -0,0 +1,80 @@
+class WorkHistoryWithBreaks
+  class BreakPlaceholder
+    def initialize(month_range:)
+      @month_range = month_range
+    end
+
+    def start_date
+      @month_range.first
+    end
+
+    def length
+      @month_range.count
+    end
+  end
+
+  def initialize(work_history)
+    @work_history = work_history.sort_by(&:start_date)
+    @work_history_with_breaks = []
+    @current_job = nil
+  end
+
+  def call
+    return [] if @work_history.empty?
+
+    @work_history.each { |job| @work_history_with_breaks << job_entry(job) }
+
+    months_in_timeline = month_range(
+      start_date: @work_history.first.start_date,
+      end_date: Time.zone.now - 1.month,
+    )
+    break_months_in_timeline = work_break_months(months_in_timeline)
+    breaks = work_breaks(break_months_in_timeline)
```

Comment:
How about method names more like this?

```
breaks = break_into_continuous_periods(
           remove_working_months(
               jobs, timeline_in_months
            )
          )
```

Reply:
Agree with `remove_working_months` but not sure about `break_into_continuous_periods`. 🤔

Reply:
Ended up renaming to `break_entries` because that's what we return in the end. What do you think?